### ISSUE TITLE
hide from players, allow players to add quests

### ISSUE THREAD

**Issue by ditzer252** [MythicPalette/simpler-quests#8]
I really how streamlined your quest tracker is.  I have a few suggestions.
For one of my games that I GM, I would prefer to keep the entire quest tracker for my eyes only.
For another game, which is more of a sandbox, it would be nice if the players could add quests themselves.
Thanks for considering.  :)

**Comment by MythicPalette** [comment]
Hey! Thanks for the suggestions. So, hiding the entire quest tracker from players, I can definitely do! I'll get to work on that today. It should be pretty simple. 

On the topic of allowing players to add/edit/delete quests... It's something I've considered but there are two things that have kind of made me hesitant.
1. allowing players to create quests means that I have to either use socketLib (which creates a dependency) or directly use the game sockets for communication.
2. That means a permission system has to be implemented in a way that still fits with the streamlined nature of my module. 
I initially decided against this feature, as I wanted to keep the module as simple as possible.

**Comment by ditzer252** [comment]
One of the main appeals of this mod is that it is “simpler” than other options currently available, so I completely understand.  Thanks!

**Comment by MythicPalette** [comment]
I just released an updated version of the quest tracker that includes an option to hide the tracker from players! Give it a look over and let me know how it works for you.

**Comment by ditzer252** [comment]
I was actually in the middle of a session when you pushed the update.  😊  It works perfectly.  Thanks!  Have a great weekend!

**Comment by SirJohn13** [comment]
Fun fact, in an earlier version players with permissions for "trusted players" (as in foundry's user management) were able to edit the quests

I like players been able to manage their quests, and that's how i was doing it.
Hope it's useful info. Maybe a simple/fitting way to implement?

Either way, i understand your reasoning.
Love this module and i appreciate your work!

**Comment by MythicPalette** [comment]
> Fun fact, in an earlier version players with permissions for "trusted players" (as in foundry's user management) were able to edit the quests
> 
> I like players been able to manage their quests, and that's how i was doing it. Hope it's useful info. Maybe a simple/fitting way to implement?
> 
> Either way, i understand your reasoning. Love this module and i appreciate your work!

Really interesting that it worked that way in the past. It definitely shouldn't have, lol. I'll play around with the idea a little more. If I can implement it without requiring any other modules and without over complicating the system, I'll definitely reconsider adding it. I make no promises, but I'm definitely willing to see if I can do it!

**Comment by MythicPalette** [comment]
I've made some headway. I managed to write my own little socket handler which will allow me to enable player edits without requiring the socketLib module. So that's good news. The system also differentiates between quests made by players and quests made by a GM. 

The system so far has three settings:
* `Players Can Create` Allow them to create quests but also to edit quests and mark objectives but only for quests not made by GM.
* `Players Can Edit GM Quests` Allow them to edit GM quests made by a GM. Warning, this will give them full access to quests in the same way that the GM has with the only exception being that they can't delete them.
* `Players Can Mark GM Quests` Allow players to mark objectives on GM quests. Great for letting players mark things done without giving them full edit access to the quest.

Allowing players to create quests will allow them to create them but also allow them to edit quests and mark objectives but only for quests not made by GM.
